### PR TITLE
Schedule SLE Micro toolbox container tests

### PR DESCRIPTION
**Pull request by jlausuch**
Includes:
  - Automation for SLEM tests which publish updated SLEM HDD
  - toolbox,pm test updated to use the toolbox image under test

Related ticket: https://progress.opensuse.org/issues/113036
VRs:
- HDD creation with system update: [slem 5.1](https://openqa.suse.de/tests/overview?distri=sle-micro&version=5.1&build=GM&groupid=377) [slem 5.2](https://openqa.suse.de/tests/overview?distri=sle-micro&version=5.2&build=GM&groupid=377)
- Toolbox test with specific image: [slem 5.1](https://openqa.suse.de/tests/9124243) [slem 5.2](https://openqa.suse.de/tests/9124242)

### CODE REVIEW

File: tests/microos/toolbox.pm [hunk 1]
```diff
@@ -39,6 +39,16 @@ sub run {
     select_console 'root-console';
     $self->create_user;
 
+    my $toolbox_image_to_test = get_var('CONTAINER_IMAGE_TO_TEST');
+
+    if ($toolbox_image_to_test) {
+        (my $registry = $toolbox_image_to_test) =~ s/\/.*//;
```
Comment (grisu48) [inline tests/microos/toolbox.pm:45]: A small comment describing what those cryptic regexes are would be very helpful for our future selves as regexes are typically write-only :wink:

File: tests/microos/toolbox.pm [hunk 2]
```diff
@@ -76,25 +86,28 @@ sub run {
...
 
-    record_info 'Test', 'Pulling toolbox image from different registry';
-    # Switch default registries for openSUSE MicroOS and SLE Micro
-    if (is_sle_micro) {
-        assert_script_run 'echo -e "REGISTRY=registry.opensuse.org\nIMAGE=opensuse/toolbox" > ~/.toolboxrc';
-        validate_script_output 'toolbox -r cat /etc/os-release', sub { m/opensuse/ }, timeout => 180;
-    } else {
-        assert_script_run 'echo -e "REGISTRY=registry.suse.com\nIMAGE=suse/sle-micro/5.0/toolbox" > ~/.toolboxrc';
-        validate_script_output 'toolbox -r cat /etc/os-release', sub { m/sles/ }, timeout => 180;
+    unless ($toolbox_image_to_test) {
```
Comment (grisu48) [inline tests/microos/toolbox.pm:89]: Why are we limiting the toolbox tests to the cases when we have a custom toolbox URI specified? Is toolbox not being tested by default also for SLEM?

**Review by grisu48 — COMMENTED** [review]
Very good work! I have two questions, but then LGTM

File: tests/microos/toolbox.pm (hunk 2 above)
Comment (jlausuch) [reply tests/microos/toolbox.pm:89]: Yes, this module is executed in regular SLEM container jobs, but using the default image which comes from registry.suse.com.

I put this `unless` here to skip this checks which are testing functionality of `toolboxrc` using different parameters, which is what we need to test the toolbox image under test..

File: tests/microos/toolbox.pm (hunk 1 above)
Comment (jlausuch) [reply tests/microos/toolbox.pm:49]: done

File: tests/microos/toolbox.pm (hunk 2 above)
Comment (grisu48) [reply tests/microos/toolbox.pm:93]: Ahh, I see. The actual tests are coming later. Thnx for the clarification!

**Review by mloviska — APPROVED** [review]

**Pull request merged**